File: StreamProcessorService/src/main/java/com/stream/processor/enrichment/ActivityLevelEnrichmentStrategy.java
package com.stream.processor.enrichment;

import com.stream.processor.Model.SensorEvent;
import org.springframework.stereotype.Component;


@Component
public class ActivityLevelEnrichmentStrategy implements EnrichmentStrategy {


    @Override
    public SensorEvent enrich(SensorEvent event) {
        if (event.getActivityLevel() != null && !event.getActivityLevel().isEmpty()) {
            return event;
        }
        return event.toBuilder().activityLevel("unknown").build();
    }
}

File: validation_service/activity_level_enrichment_strategy.py
"""Enriches event with default activity level if missing"""
from typing import Dict, Any


class ActivityLevelEnrichmentStrategy:
    """Enriches event with default activity level if missing"""
    
    def enrich(self, event: Dict[str, Any]) -> Dict[str, Any]:
        if event.get('activity_level'):
            return event
        event = event.copy()
        event['activity_level'] = 'unknown'
        return event

